ros2-library/src/main/java/us/ihmc/ros2/ROS2NodeBuilder.java

```diff
@@ -0,0 +1,260 @@
+      return this;
+   }
+
+   private ParticipantProfile buildProfile()
+   {
+      ParticipantProfile profile = ParticipantProfile.create();
+
+      // Set up ROS Domain ID
+      {
+         if (domainIDValid(domainId))
+         {
+            LogTools.info("Using a programmatically set ROS Domain ID: {}", domainId);
+         }
+         else
+         {
+            // Try to find a ROS Domain ID
+            domainId = findDomainID(parseNetworkParametersConfig, parseProperties, parseEnvironment);
+
+            // If a valid domain ID was not found automatically
+            if (!domainIDValid(domainId))
+            {
+               domainId = 0;
+
+               LogTools.warn("Unable to find any ROS Domain ID");
+               LogTools.warn(
+                     "You can set a ROS Domain ID via: system property (ROS_DOMAIN_ID), environment variable (ROS_DOMAIN_ID), RTPSDomainID in IHMCNetworkParameters.ini");
+               LogTools.warn("Using a default ROS Domain ID of {}", domainId);
+            }
+         }
+
+         profile.domainId(domainId);
+      }
+
+      // Set up transports
+      {
+         profile.useBuiltinTransports(false);
+
+         if (useSharedMemory)
+            profile.addSharedMemoryTransport();
+
+         profile.addUDPv4Transport(addressRestriction);
+
+         if (transportSpecial != null)
+         {
+            switch (transportSpecial)
+            {
+               case SHARED_MEMORY_ONLY ->
+               {
+                  profile.useOnlySharedMemoryTransport();
+               }
+               case LOOPBACK_ADDRESS_ONLY ->
+               {
+                  InetAddress loopbackAddress = InetAddress.getLoopbackAddress();
+
+                  addressRestriction = new InetAddress[] {loopbackAddress};
+               }
+               case UDPV4_ONLY ->
+               {
+                  profile.useOnlyUDPv4Transport(addressRestriction);
+               }
+            }
+         }
+      }
+
+      // Print the current transports
+
+      return profile;
+   }
+
+   public ROS2Node build(String name)
+   {
+      return new ROS2Node(name, namespace, buildProfile());
+   }
+
+   public RealtimeROS2Node buildRealtime(String name)
+   {
+      return buildRealtime(name, new PeriodicNonRealtimeThreadSchedulerFactory());
+   }
+
+   public RealtimeROS2Node buildRealtime(String name, PeriodicThreadSchedulerFactory threadFactory)
+   {
+      return new RealtimeROS2Node(name, namespace, buildProfile(), threadFactory);
+   }
+
+   protected static boolean domainIDValid(int domainID)
+   {
+      return domainID >= 0 && domainID <= 232;
+   }
+
+   /**
+    * Find the ROS Domain ID from either:
+    *    - IHMCNetworkParameters.ini
+    *    - System property
+    *    - Environment variable
+    * prioritized in that order
+    */
+   protected static int findDomainID(boolean parseNetworkParametersConfig, boolean parseProperties, boolean parseEnvironment)
```

Comment:
It might be nice if we combined all the prints into one line that also explains the priority, for examples:
```
[INFO] .... ROS_DOMAIN_ID=13 -> -Dros.domain.id=52 -> ~/.ihmc/IHMCNetworkParameters.ini:RTPSDomainID=123
[INFO] .... ROS_DOMAIN_ID=11 -> ~/.ihmc/IHMCNetworkParameters.ini:RTPSDomainID=213
[INFO] .... -Dros.domain.id=35
```

Reply:
done